selfdrive/modeld/modeld.py

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env python3
+import sys
+import time
+import logging
+import numpy as np
+from pathlib import Path
+from typing import Dict, Optional
+from setproctitle import setproctitle  # pylint: disable=no-name-in-module
+from cereal.messaging import PubMaster, SubMaster
+from cereal.visionipc import VisionIpcClient, VisionStreamType, VisionBuf
+from system.hardware import PC
+from common.params import Params
+from common.filter_simple import FirstOrderFilter
+from common.realtime import set_core_affinity, set_realtime_priority
+from selfdrive.modeld.models.commonmodel_pyx import ModelFrame, CLContext, Runtime # pylint: disable=import-error, no-name-in-module
+from selfdrive.modeld.models.driving_pyx import USE_THNEED, PublishState # pylint: disable=import-error, no-name-in-module
+from selfdrive.modeld.models.driving_pyx import create_model_msg, create_pose_msg, update_calibration # pylint: disable=import-error, no-name-in-module
+
+if USE_THNEED:
+  from selfdrive.modeld.runners.thneedmodel_pyx import ThneedModel as ModelRunner # pylint: disable=import-error, no-name-in-module
+else:
+  from selfdrive.modeld.runners.onnxmodel_pyx import ONNXModel as ModelRunner # pylint: disable=import-error, no-name-in-module
+
+FEATURE_LEN = 128
```

Comment:
these constants shouldn't be duplicated. if we can't have cython handle them, we can generate a header file at build time